servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -367,8 +368,16 @@ public boolean isDirectory(long id) {
    */
   public long getFileId(TachyonURI path) throws InvalidPathException {
     synchronized (mInodeTree) {
-      Inode inode = mInodeTree.getInodeByPath(path);
-      return inode.getId();
+      try {
+        Inode inode = mInodeTree.getInodeByPath(path);
```

Comment:
I don't like the fact that we are using exceptions for flow control. 
We may want to have a check if the path exists in Tachyon space rather than expecting exception on this.